tools/run_tests/xds_k8s_test_driver/bin/cleanup/cleanup.py

```diff
@@ -0,0 +1,253 @@
+    exec_gcloud(project, 'compute', 'target-http-proxies', 'delete',
+                f'test-target-proxy{suffix}')
+    exec_gcloud(project, 'alpha', 'compute', 'target-grpc-proxies', 'delete',
+                f'test-target-proxy{suffix}')
+    exec_gcloud(project, 'compute', 'url-maps', 'delete', f'test-map{suffix}')
+    exec_gcloud(project, 'compute', 'backend-services', 'delete',
+                f'test-backend-service{suffix}', '--global')
+    exec_gcloud(project, 'compute', 'backend-services', 'delete',
+                f'test-backend-service-alternate{suffix}', '--global')
+    exec_gcloud(project, 'compute', 'backend-services', 'delete',
+                f'test-backend-service-extra{suffix}', '--global')
+    exec_gcloud(project, 'compute', 'backend-services', 'delete',
+                f'test-backend-service-more-extra{suffix}', '--global')
+    exec_gcloud(project, 'compute', 'firewall-rules', 'delete', f'test-fw-rule{suffix}')
+    exec_gcloud(project, 'compute', 'health-checks', 'delete', f'test-hc{suffix}')
+    exec_gcloud(project, 'compute', 'instance-groups', 'managed', 'delete',
+                f'test-ig{suffix}', '--zone', ZONE)
+    exec_gcloud(project, 'compute', 'instance-groups', 'managed', 'delete',
+                f'test-ig-same-zone{suffix}', '--zone', ZONE)
+    exec_gcloud(project, 'compute', 'instance-groups', 'managed', 'delete',
+                f'test-ig-secondary-zone{suffix}', '--zone', SECONDARY_ZONE)
+    exec_gcloud(project, 'compute', 'instance-templates', 'delete',
+                f'test-template{suffix}')
+
+
+# cleanup_td creates TrafficDirectorManager (and its varients for security and
+# AppNet), and then calls the cleanup() methods.
+#
+# Note that the varients are all based on the basic TrafficDirectorManager, so
+# their `cleanup()` might do duplicate work. But deleting an non-exist resource
+# returns 404, and is OK.
+def cleanup_td_for_gke(project, network, resource_prefix, resource_suffix):
+    gcp_api_manager = gcp.api.GcpApiManager()
+    plain_td = traffic_director.TrafficDirectorManager(
+        gcp_api_manager,
+        project=project,
+        network=network,
+        resource_prefix=resource_prefix,
+        resource_suffix=resource_suffix)
+    security_td = traffic_director.TrafficDirectorSecureManager(
+        gcp_api_manager,
+        project=project,
+        network=network,
+        resource_prefix=resource_prefix,
+        resource_suffix=resource_suffix)
+    # TODO: cleanup appnet resources.
+    # appnet_td = traffic_director.TrafficDirectorAppNetManager(
+    #     gcp_api_manager,
+    #     project=project,
+    #     network=network,
+    #     resource_prefix=resource_prefix,
+    #     resource_suffix=resource_suffix)
+
+    logger.info('----- Removing traffic director for gke, prefix %s, suffix %s', resource_prefix, resource_suffix)
+    security_td.cleanup(force=True)
+    # appnet_td.cleanup(force=True)
+    plain_td.cleanup(force=True)
```

Comment:
I will recommend to consider also cleaning GKE resources. When Kokoro job times out, or we made a big mistake in framework, the K8s Namespace might be left behind. If that accumulates terribly, the cluster resource could deplete.

Reply:
Done